### PR TITLE
NOISSUE - Test Covid19 on Cocos

### PR DESCRIPTION
**Pull request by rodneyosodo**
This pull request refactors the COVID-19 detection model code to enhance modularity and readability, adds support for Unix socket-based inference, and updates the documentation with comprehensive setup and usage instructions.

- **Enhancements**:
    - Refactored `covid19.py` to support both model training and inference via a Unix socket.
    - Updated `predict.py` to allow loading model weights from a binary file generated by Cocos.
    - Improved dataset preparation and training logic in `covid19.py` for better modularity and readability.
- **Documentation**:
    - Expanded `README.md` with detailed setup instructions, including virtual environment setup, dataset preparation, model training, and testing with Cocos.
- **Bug Fixes**
	- Improved image selection logic and error handling in model preparation.
- **Chores**
	- Updated `.gitignore` to streamline file tracking.
	- Added `matplotlib` as a new dependency in the requirements.

### CODE REVIEW

File: covid19/README.md [hunk 1]
```diff
@@ -1,31 +1,259 @@
...
 ```
 
-### Test Model
+To run the model on cocos, we would need to send the results to a unix socket. This can be done by running:
+
+```bash
+python covid19.py /tmp/covid19.sock data/h1 data/h2 data/h3
+```
+
+The folders `data/h1`, `data/h1` or `data/h1` can be passed as zipped files (after they have been zipped) for example:
```
Comment (SammyOina) [inline covid19/README.md:50]: add documentation on zipping the directories correctly

**Review by SammyOina — CHANGES_REQUESTED** [review]

File: covid19/README.md [hunk 2]
```diff
@@ -1,31 +1,259 @@
...
 
-### Train Model
 `covid19.py` is a model that needs to be trained.
 
-Do do the training, execute:
+To do the training, execute:
 
 ```bash
- python covid19.py data/h1 data/h2 data/h3 --model model.pth  
+python covid19.py --model model.pth data/h1 data/h2 data/h3
```
Comment (SammyOina) [inline covid19/README.md:41]: this fails

```shell
python covid19.py --model model.pth data/h1 data/h2 data/h3                                                                                                                                           1 ✘  mypy  
Traceback (most recent call last):
  File "/home/sammyk/Documents/ai/covid19/covid19.py", line 216, in <module>
    main()
  File "/home/sammyk/Documents/ai/covid19/covid19.py", line 161, in main
    with zipfile.ZipFile(hospital, "r") as zip_ref:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/zipfile/__init__.py", line 1331, in __init__
    self.fp = io.open(file, filemode)
              ^^^^^^^^^^^^^^^^^^^^^^^
IsADirectoryError: [Errno 21] Is a directory: 'data/h1'
```

**Review by SammyOina — CHANGES_REQUESTED** [review]

File: covid19/README.md [hunk 3]
```diff
@@ -1,31 +1,259 @@
...
\ No newline at end of file
+python predict.py --model model.pth --image data/test/COVID/COVID-2.png
+```
+
+## Testing with Cocos
+
+Currently, the most suitable way to test it with cocos is to use the virtual machine that has AMDSEV enabled. Login to the virtual machine and run the following commands:
+
+```bash
+mkdir -p /tmp/covid19-test && cd /tmp/covid19-test
```
Comment (WashingtonKK) [inline covid19/README.md:69]: Testing on this folder results in this error:

```shell
&{message:"                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/tmp/algorithm3912989560\", line 53, in __init__\n    self.images[class_name] = get_images(class_name)\n                              ^^^^^^^^^^^^^^^^^^^^^^\n  File \"/tmp/algorithm3912989560\", line 44, in get_images\n    for x in os.listdir(dir_path)\n             ^^^^^^^^^^^^^^^^^^^^\n"  computation_id:"1"  level:"ERROR"  timestamp:{seconds:1721858799  nanos:921433577}}
received agent log
&{message:"FileNotFoundError: [Errno 2] No such file or directory: '/tmp/work/data/Normal'\n"  computation_id:"1"  level:"ERROR"  timestamp:{seconds:1721858799  nanos:921444087}}
```

File: covid19/predict.py [hunk 4]
```diff
@@ -1,59 +1,93 @@
...
     image_path = args.image
 
-    class_names = ['Normal', 'Viral Pneumonia', 'COVID']
+    class_names = ["Normal", "Viral Pneumonia", "COVID"]
 
     model = load_model(model_path, class_names)
     predicted_class = predict(model, image_path, class_names)
-    print(f'The predicted class for the image is: {predicted_class}')
+    print(f"The predicted class for the image is: {predicted_class}")
     show_image_with_prediction(image_path, predicted_class)
```
Comment (SammyOina) [inline covid19/predict.py:89]: write to file instead in results directory

**Review by SammyOina — CHANGES_REQUESTED** [review]

File: covid19/README.md [hunk 5]
```diff
@@ -1,31 +1,248 @@
...
 Inference can be done using `predict.py`:
 
 ```bash
-python predict.py --model model.pth --image test/COVID/COVID-3.png
-```
\ No newline at end of file
+python predict.py --model results/model.pth --image data/test/COVID/COVID-2.png
+```
+
+## Testing with Cocos
```
Comment (SammyOina) [inline covid19/README.md:53]: also add testing with prism

**Review by SammyOina — CHANGES_REQUESTED** [review]

**Review by SammyOina — APPROVED** [review]

**Pull request merged**